hack/lib/golang.sh

```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# shellcheck disable=SC2034 # Variables sourced in other scripts.
```

Comment:
When shellcheck disable statements are placed here they apply to the whole file (since this applies to several lines)

Reply:
on the fence about this, we should probably be verifying which we export and which are unused and remove things that are unused.